quickwit-search/src/fetch_docs.rs

```diff
@@ -144,30 +163,140 @@ async fn get_searcher_for_split_without_cache(
     Ok(reader)
 }
 
+/// A struct for holding a fetched document's content and highlights.
+struct DocBundle {
```

Comment:
Not fan `DocBundle` name. Maybe `Document` with `content_json` and `snippets_json`?